In [1]:
import pandas as pd
pd.options.display.max_columns = None
import re

In [2]:
df = pd.read_csv("union_datos.csv", index_col= "index")

In [4]:
df.columns

Index(['age', 'gender', 'pais', 'estudios', 'puesto_trabajo', 'experiencia',
       'lenguaje_recomendado', 'experiencia_ML', 'industria', 'tamaño_empresa',
       ...
       'Q38_B_Part_3', 'Q38_B_Part_4', 'Q38_B_Part_5', 'Q38_B_Part_6',
       'Q38_B_Part_7', 'Q38_B_Part_8', 'Q38_B_Part_9', 'Q38_B_Part_10',
       'Q38_B_Part_11', 'Q38_B_OTHER'],
      dtype='object', length=150)

In [3]:
df.head(3)

age gender       pais           estudios           puesto_trabajo  \
index                                                                        
0      50-54    Man      India  Bachelor’s degree                    Other   
1      50-54    Man  Indonesia    Master’s degree  Program/Project Manager   
2      22-24    Man   Pakistan    Master’s degree        Software Engineer   

      experiencia lenguaje_recomendado                         experiencia_ML  \
index                                                                           
0      5-10 years               Python                             5-10 years   
1       20+ years               Python                               < 1 year   
2       1-3 years               Python  I do not use machine learning methods   

                       industria        tamaño_empresa        salario  \
index                                                                   
0      Manufacturing/Fabrication      50-249 employees  25,000-29,999   
1      Manufacturing/Fabrication  1000-9,999 employees  60,000-69,999   
2            Academics/Education  1000-9,999 employees         $0-999   

                                          herramienta_DA  lenguaje_utilizado  \
index                                                                          
0      Local development environments (RStudio, Jupyt...           Python, R   
1        Advanced statistical software (SPSS, SAS, etc.)   SQL, C, C++, Java   
2      Basic statistical software (Microsoft Excel, G...   Python, C++, Java   

                                    IDE  \
index                                     
0                           Vim / Emacs   
1            Notepad++,Jupyter Notebook   
2      PyCharm ,Jupyter Notebook, Other   

                              herramientas_visualizacion  \
index                                                      
0       Matplotlib ,Seaborn ,Ggplot / ggplot2 ,Shiny ...   
1                                            Matplotlib    
2                                            Matplotlib    

                                     Q16  \
index                                      
0      Scikit-learn , TensorFlow ,Caret    
1                          Scikit-learn    
2                                    NaN   

                                                     Q17               Q32  \
index                                                                        
0      Linear or Logistic Regression, Decision Trees ...       PostgreSQL    
1      Linear or Logistic Regression, Decision Trees ...               NaN   
2                                                    NaN  MySQL , MongoDB    

         Q34             kaggle            colab ninguno Q27_A_Part_1  \
index                                                                   
0       None                NaN  Colab Notebooks     NaN          NaN   
1        NaN   Kaggle Notebooks  Colab Notebooks     NaN          NaN   
2       None   Kaggle Notebooks              NaN     NaN          NaN   

      Q27_A_Part_2                   Q27_A_Part_3 Q27_A_Part_4 Q27_A_Part_5  \
index                                                                         
0              NaN   Google Cloud Platform (GCP)           NaN          NaN   
1              NaN                            NaN          NaN          NaN   
2              NaN                            NaN          NaN          NaN   

      Q27_A_Part_6 Q27_A_Part_7 Q27_A_Part_8 Q27_A_Part_9 Q27_A_Part_10  \
index                                                                     
0              NaN          NaN          NaN          NaN           NaN   
1              NaN          NaN          NaN          NaN           NaN   
2              NaN          NaN          NaN          NaN           NaN   

      Q27_A_Part_11 Q27_A_OTHER  Q28 Q39_Part_1 Q39_Part_2 Q39_Part_3  \
index                                                                   
0               NaN         NaN 

In [5]:
#Eliminamos las columnas Q27_A
df.drop(['Q27_A_Part_1', 'Q27_A_Part_2','Q27_A_Part_3','Q27_A_Part_4','Q27_A_Part_5','Q27_A_Part_6','Q27_A_Part_7','Q27_A_Part_8','Q27_A_Part_9','Q27_A_Part_10','Q27_A_Part_11','Q27_A_OTHER'], axis = 1, inplace = True)
df.head(2)

age gender       pais           estudios           puesto_trabajo  \
index                                                                        
0      50-54    Man      India  Bachelor’s degree                    Other   
1      50-54    Man  Indonesia    Master’s degree  Program/Project Manager   

      experiencia lenguaje_recomendado experiencia_ML  \
index                                                   
0      5-10 years               Python     5-10 years   
1       20+ years               Python       < 1 year   

                       industria        tamaño_empresa        salario  \
index                                                                   
0      Manufacturing/Fabrication      50-249 employees  25,000-29,999   
1      Manufacturing/Fabrication  1000-9,999 employees  60,000-69,999   

                                          herramienta_DA  lenguaje_utilizado  \
index                                                                          
0      Local development environments (RStudio, Jupyt...           Python, R   
1        Advanced statistical software (SPSS, SAS, etc.)   SQL, C, C++, Java   

                               IDE  \
index                                
0                      Vim / Emacs   
1       Notepad++,Jupyter Notebook   

                              herramientas_visualizacion  \
index                                                      
0       Matplotlib ,Seaborn ,Ggplot / ggplot2 ,Shiny ...   
1                                            Matplotlib    

                                     Q16  \
index                                      
0      Scikit-learn , TensorFlow ,Caret    
1                          Scikit-learn    

                                                     Q17           Q32    Q34  \
index                                                                           
0      Linear or Logistic Regression, Decision Trees ...   PostgreSQL    None   
1      Linear or Logistic Regression, Decision Trees ...           NaN    NaN   

              Q10_kaggle        Q10_colab Q10_none  Q28 Q39_Part_1 Q39_Part_2  \
index                                                                           
0                    NaN  Colab Notebooks      NaN  NaN        NaN        NaN   
1       Kaggle Notebooks  Colab Notebooks      NaN  NaN        NaN        NaN   

      Q39_Part_3 Q39_Part_4 Q39_Part_5 Q39_Part_6 Q39_Part_7 Q39_Part_8  \
index                                                                     
0            NaN    GitHub         NaN    Kaggle         NaN        NaN   
1            NaN        NaN        NaN        NaN        NaN        NaN   

      Q39_Part_9 Q39_OTHER Q40_Part_1 Q40_Part_2            Q40_Part_3  \
index                                                                    
0            NaN       NaN   Coursera        edX  Kaggle Learn Courses   
1            NaN       NaN        NaN        NaN  Kaggle Learn Courses   

      Q40_Part_4 Q40_Part_5 Q40_Part_6 Q40_Part_7 Q40_Part_8  \
index                                                          
0       DataCamp        NaN    Udacity      Udemy        NaN   
1            NaN        NaN        NaN        NaN        NaN   

                                              Q40_Part_9  \
index                                                      
0                                                    NaN   
1      Cloud-certification programs (direct from AWS,...   

                                             Q40_Part_10 Q40_Part_11  \
index                                                                  
0                                                    NaN         NaN   
1      University Courses (resulting in a university ...         NaN   

      Q40_OTHER Q27_B_Part_1 Q27_B_Part_2                   Q27_B_Part_3  \
index                                                                      
0           NaN          NaN          NaN                            NaN   
1           NaN    

---

### Renombrado columnas

---

In [5]:
no_nulos_q39 = df.loc[:,'Q39_Part_1':'Q39_OTHER'].notnull().sum()/df.shape[0]*100 # Porcentaje de no nulos en la pregunta Q39 - Quedarnos solo con las más representativas

In [6]:
for col in no_nulos_q39.index:
    print (col, '--->', df[col].value_counts().index[0], '--->', (df.loc[:,col].notnull().sum()/df.shape[0]*100).round(2),'%' )

Q39_Part_1 --->  Plotly Dash  ---> 1.13 %
Q39_Part_2 --->  Streamlit  ---> 1.49 %
Q39_Part_3 --->  NBViewer  ---> 0.52 %
Q39_Part_4 --->  GitHub  ---> 17.66 %
Q39_Part_5 --->  Personal blog  ---> 2.71 %
Q39_Part_6 --->  Kaggle  ---> 11.8 %
Q39_Part_7 --->  Colab  ---> 7.12 %
Q39_Part_8 --->  Shiny  ---> 1.17 %
Q39_Part_9 ---> I do not share my work publicly ---> 12.19 %
Q39_OTHER ---> Other ---> 1.18 %


### Q39 donde compartes publicamente tu trabajo
    Q39_Part_1 --->  Plotly Dash  ---> 1.13 %
    Q39_Part_2 --->  Streamlit  ---> 1.49 %
    Q39_Part_3 --->  NBViewer  ---> 0.52 %
    Q39_Part_4 --->  GitHub  ---> 17.66 %
    Q39_Part_5 --->  Personal blog  ---> 2.71 %
    Q39_Part_6 --->  Kaggle  ---> 11.8 %
    Q39_Part_7 --->  Colab  ---> 7.12 %
    Q39_Part_8 --->  Shiny  ---> 1.17 %
    Q39_Part_9 ---> I do not share my work publicly ---> 12.19 %
    Q39_OTHER ---> Other ---> 1.18 %

Nos quedamos con las columnas más representativas: GitHub, I do not share my work publicly, Kaggle, Colab

In [7]:
df.drop(['Q39_Part_1', 'Q39_Part_2','Q39_Part_3','Q39_Part_5','Q39_Part_8','Q39_OTHER'], axis = 1, inplace = True)

In [8]:
#Cambiamos el nombre a las columnas de Q39
col_q39 = {'Q39_Part_4': 'Q39_GitHub', 'Q39_Part_9': 'Q39_None', 'Q39_Part_6': 'Q39_Kaggle','Q39_Part_7': 'Q39_Colab' }
df.rename(columns = col_q39, inplace = True)

In [9]:
df.head(2)

age gender       pais           estudios           puesto_trabajo  \
index                                                                        
0      50-54    Man      India  Bachelor’s degree                    Other   
1      50-54    Man  Indonesia    Master’s degree  Program/Project Manager   

      experiencia lenguaje_recomendado experiencia_ML  \
index                                                   
0      5-10 years               Python     5-10 years   
1       20+ years               Python       < 1 year   

                       industria        tamaño_empresa        salario  \
index                                                                   
0      Manufacturing/Fabrication      50-249 employees  25,000-29,999   
1      Manufacturing/Fabrication  1000-9,999 employees  60,000-69,999   

                                          herramienta_DA  lenguaje_utilizado  \
index                                                                          
0      Local development environments (RStudio, Jupyt...           Python, R   
1        Advanced statistical software (SPSS, SAS, etc.)   SQL, C, C++, Java   

                               IDE  \
index                                
0                      Vim / Emacs   
1       Notepad++,Jupyter Notebook   

                              herramientas_visualizacion  \
index                                                      
0       Matplotlib ,Seaborn ,Ggplot / ggplot2 ,Shiny ...   
1                                            Matplotlib    

                                     Q16  \
index                                      
0      Scikit-learn , TensorFlow ,Caret    
1                          Scikit-learn    

                                                     Q17           Q32    Q34  \
index                                                                           
0      Linear or Logistic Regression, Decision Trees ...   PostgreSQL    None   
1      Linear or Logistic Regression, Decision Trees ...           NaN    NaN   

                  kaggle            colab ninguno  Q28 Q39_GitHub Q39_Kaggle  \
index                                                                          
0                    NaN  Colab Notebooks     NaN  NaN    GitHub     Kaggle    
1       Kaggle Notebooks  Colab Notebooks     NaN  NaN        NaN        NaN   

      Q39_Colab Q39_None Q40_Part_1 Q40_Part_2            Q40_Part_3  \
index                                                                  
0           NaN      NaN   Coursera        edX  Kaggle Learn Courses   
1           NaN      NaN        NaN        NaN  Kaggle Learn Courses   

      Q40_Part_4 Q40_Part_5 Q40_Part_6 Q40_Part_7 Q40_Part_8  \
index                                                          
0       DataCamp        NaN    Udacity      Udemy        NaN   
1            NaN        NaN        NaN        NaN        NaN   

                                              Q40_Part_9  \
index                                                      
0                                                    NaN   
1      Cloud-certification programs (direct from AWS,...   

                                             Q40_Part_10 Q40_Part_11  \
index                                                                  
0                                                    NaN         NaN   
1      University Courses (resulting in a university ...         NaN   

      Q40_OTHER Q27_B_Part_1 Q27_B_Part_2                   Q27_B_Part_3  \
index                                                                      
0           NaN          NaN          NaN                            NaN   
1           NaN          NaN          NaN   Google Cloud Platform (GCP)    

      Q27_B_Part_4    Q27_B_Part_5 Q27_B_Part_6 Q27_B_Part_7 Q27_B_Part_8  \
index                                                                       
0              NaN             NaN          NaN          NaN          NaN   
1              NaN   Ora

In [32]:
no_nulos_q40 = df.loc[:,'Q40_Part_1':'Q40_OTHER'].notnull().sum()/df.shape[0]*100 # Porcentaje de no nulos en la pregunta Q40
no_nulos_q40.index


Index(['Q40_Part_1', 'Q40_Part_2', 'Q40_Part_3', 'Q40_Part_4', 'Q40_Part_5',
       'Q40_Part_6', 'Q40_Part_7', 'Q40_Part_8', 'Q40_Part_9', 'Q40_Part_10',
       'Q40_Part_11', 'Q40_OTHER'],
      dtype='object')

In [11]:
for col in no_nulos_q40.index:
    print (col, '--->', df[col].value_counts().index[0], '--->', (df.loc[:,col].notnull().sum()/df.shape[0]*100).round(2),'%' )

Q40_Part_1 ---> Coursera ---> 37.15 %
Q40_Part_2 ---> edX ---> 11.66 %
Q40_Part_3 ---> Kaggle Learn Courses ---> 33.38 %
Q40_Part_4 ---> DataCamp ---> 15.39 %
Q40_Part_5 ---> Fast.ai ---> 4.45 %
Q40_Part_6 ---> Udacity ---> 10.79 %
Q40_Part_7 ---> Udemy ---> 24.79 %
Q40_Part_8 ---> LinkedIn Learning ---> 10.37 %
Q40_Part_9 ---> Cloud-certification programs (direct from AWS, Azure, GCP, or similar) ---> 6.95 %
Q40_Part_10 ---> University Courses (resulting in a university degree) ---> 18.8 %
Q40_Part_11 ---> None ---> 6.78 %
Q40_OTHER ---> Other ---> 8.01 %


### Q40 en que plataformas has realizado cursos de data science
    Q40_Part_1 ---> Coursera ---> 37.15 %
    Q40_Part_2 ---> edX ---> 11.66 %
    Q40_Part_3 ---> Kaggle Learn Courses ---> 33.38 %
    Q40_Part_4 ---> DataCamp ---> 15.39 %
    Q40_Part_5 ---> Fast.ai ---> 4.45 %
    Q40_Part_6 ---> Udacity ---> 10.79 %
    Q40_Part_7 ---> Udemy ---> 24.79 %
    Q40_Part_8 ---> LinkedIn Learning ---> 10.37 %
    Q40_Part_9 ---> Cloud-certification programs (direct from AWS, Azure, GCP, or similar) ---> 6.95 %
    Q40_Part_10 ---> University Courses (resulting in a university degree) ---> 18.8 %
    Q40_Part_11 ---> None ---> 6.78 %
    Q40_OTHER ---> Other ---> 8.01 %


In [29]:
lista =[]
for col in no_nulos_q40.index:
    plat = df[col].value_counts().index[0]
    lista.append(re.findall('Q\d{2}',col)[0]+'_'+plat.replace(' ','_').lower())
lista

['Q40_coursera',
 'Q40_edx',
 'Q40_kaggle_learn_courses',
 'Q40_datacamp',
 'Q40_fast.ai',
 'Q40_udacity',
 'Q40_udemy',
 'Q40_linkedin_learning',
 'Q40_cloud-certification_programs_(direct_from_aws,_azure,_gcp,_or_similar)',
 'Q40_university_courses_(resulting_in_a_university_degree)',
 'Q40_none',
 'Q40_other']

In [30]:
col_q40 = dict(zip(no_nulos_q40.index, lista))
col_q40


{'Q40_Part_1': 'Q40_coursera',
 'Q40_Part_2': 'Q40_edx',
 'Q40_Part_3': 'Q40_kaggle_learn_courses',
 'Q40_Part_4': 'Q40_datacamp',
 'Q40_Part_5': 'Q40_fast.ai',
 'Q40_Part_6': 'Q40_udacity',
 'Q40_Part_7': 'Q40_udemy',
 'Q40_Part_8': 'Q40_linkedin_learning',
 'Q40_Part_9': 'Q40_cloud-certification_programs_(direct_from_aws,_azure,_gcp,_or_similar)',
 'Q40_Part_10': 'Q40_university_courses_(resulting_in_a_university_degree)',
 'Q40_Part_11': 'Q40_none',
 'Q40_OTHER': 'Q40_other'}

In [33]:
col_q40 = {'Q40_Part_1': 'Q40_coursera',
 'Q40_Part_2': 'Q40_edx',
 'Q40_Part_3': 'Q40_kaggle_learn_courses',
 'Q40_Part_4': 'Q40_datacamp',
 'Q40_Part_5': 'Q40_fast.ai',
 'Q40_Part_6': 'Q40_udacity',
 'Q40_Part_7': 'Q40_udemy',
 'Q40_Part_8': 'Q40_linkedin_learning',
 'Q40_Part_9': 'Q40_cloud-certification',
 'Q40_Part_10': 'Q40_university_courses',
 'Q40_Part_11': 'Q40_none',
 'Q40_OTHER': 'Q40_other'}
df.rename(columns = col_q40, inplace = True)

In [35]:
df.head(2)

age gender       pais           estudios           puesto_trabajo  \
index                                                                        
0      50-54    Man      India  Bachelor’s degree                    Other   
1      50-54    Man  Indonesia    Master’s degree  Program/Project Manager   

      experiencia lenguaje_recomendado experiencia_ML  \
index                                                   
0      5-10 years               Python     5-10 years   
1       20+ years               Python       < 1 year   

                       industria        tamaño_empresa        salario  \
index                                                                   
0      Manufacturing/Fabrication      50-249 employees  25,000-29,999   
1      Manufacturing/Fabrication  1000-9,999 employees  60,000-69,999   

                                          herramienta_DA  lenguaje_utilizado  \
index                                                                          
0      Local development environments (RStudio, Jupyt...           Python, R   
1        Advanced statistical software (SPSS, SAS, etc.)   SQL, C, C++, Java   

                               IDE  \
index                                
0                      Vim / Emacs   
1       Notepad++,Jupyter Notebook   

                              herramientas_visualizacion  \
index                                                      
0       Matplotlib ,Seaborn ,Ggplot / ggplot2 ,Shiny ...   
1                                            Matplotlib    

                                     Q16  \
index                                      
0      Scikit-learn , TensorFlow ,Caret    
1                          Scikit-learn    

                                                     Q17           Q32    Q34  \
index                                                                           
0      Linear or Logistic Regression, Decision Trees ...   PostgreSQL    None   
1      Linear or Logistic Regression, Decision Trees ...           NaN    NaN   

                  kaggle            colab ninguno  Q28 Q39_GitHub Q39_Kaggle  \
index                                                                          
0                    NaN  Colab Notebooks     NaN  NaN    GitHub     Kaggle    
1       Kaggle Notebooks  Colab Notebooks     NaN  NaN        NaN        NaN   

      Q39_Colab Q39_None Q40_coursera Q40_edx Q40_kaggle_learn_courses  \
index                                                                    
0           NaN      NaN     Coursera     edX     Kaggle Learn Courses   
1           NaN      NaN          NaN     NaN     Kaggle Learn Courses   

      Q40_datacamp Q40_fast.ai Q40_udacity Q40_udemy Q40_linkedin_learning  \
index                                                                        
0         DataCamp         NaN     Udacity     Udemy                   NaN   
1              NaN         NaN         NaN       NaN                   NaN   

                                 Q40_cloud-certification  \
index                                                      
0                                                    NaN   
1      Cloud-certification programs (direct from AWS,...   

                                  Q40_university_courses Q40_none Q40_other  \
index                                                                         
0                                                    NaN      NaN       NaN   
1      University Courses (resulting in a university ...      NaN       NaN   

      Q27_B_Part_1 Q27_B_Part_2                   Q27_B_Part_3 Q27_B_Part_4  \
index                                                                         
0              NaN          NaN                            NaN          NaN   
1              NaN          NaN   Google Cloud Platform (GCP)           NaN   

         Q27_B_Part_5 Q27_B_Part_6 Q27_B_Part_7 Q27_B_Part_8 Q27_B_Part_9  \
index                                                                     

---

In [ ]:
df['Q31_B_Part_1'].value_counts().index[0] #Sacamos el nombre de la columna Q31_B_part_1 para ver si podemos automatizar el cambio de nombres

' Amazon SageMaker '

In [ ]:
df.loc[:,'Q29_B_Part_1':'Q29_B_OTHER'].notnull().sum()/df.shape[0]*100 # Porcentaje de no nulos en la pregunta Q29

Q29_B_Part_1    20.706118
Q29_B_Part_2    22.303931
Q29_B_Part_3    28.864590
Q29_B_Part_4     3.734648
Q29_B_OTHER      0.300312
dtype: float64

In [ ]:
df.loc[:,'Q27_B_Part_1':'Q27_B_OTHER'].notnull().sum()/df.shape[0]*100 # Porcentaje de no nulos en la pregunta Q27

Q27_B_Part_1     28.853040
Q27_B_Part_2     21.745659
Q27_B_Part_3     28.814538
Q27_B_Part_4      9.968044
Q27_B_Part_5      6.591460
Q27_B_Part_6      3.503638
Q27_B_Part_7      3.018519
Q27_B_Part_8      3.049320
Q27_B_Part_9      2.433296
Q27_B_Part_10     2.117584
Q27_B_Part_11     8.616640
Q27_B_OTHER       1.035691
dtype: float64

In [ ]:
df.loc[:,'Q39_Part_1':'Q39_OTHER'].notnull().sum()/df.shape[0]*100 # Porcentaje de no nulos en la pregunta Q39 - Quedarnos solo con las más representativas

Q39_Part_1     1.128095
Q39_Part_2     1.490009
Q39_Part_3     0.523621
Q39_Part_4    17.656797
Q39_Part_5     2.714357
Q39_Part_6    11.800716
Q39_Part_7     7.115081
Q39_Part_8     1.174296
Q39_Part_9    12.193432
Q39_OTHER      1.178147
dtype: float64

In [ ]:
df.loc[:,'Q40_Part_1':'Q40_OTHER'].notnull().sum()/df.shape[0]*100 # Porcentaje de no nulos en la pregunta Q40

Q40_Part_1     37.150117
Q40_Part_2     11.658261
Q40_Part_3     33.380819
Q40_Part_4     15.385208
Q40_Part_5      4.454626
Q40_Part_6     10.791976
Q40_Part_7     24.794979
Q40_Part_8     10.368460
Q40_Part_9      6.945674
Q40_Part_10    18.796442
Q40_Part_11     6.783968
Q40_OTHER       8.012166
dtype: float64

In [ ]:
# Q27 no nos aporta info extra - Q28 nos aporta suficiente info
df.loc[:,'Q27_A_Part_1':'Q27_A_OTHER'].isnull().sum()/df.shape[0]*100 

Q27_A_Part_1     85.673584
Q27_A_Part_2     90.567127
Q27_A_Part_3     87.902822
Q27_A_Part_4     97.797713
Q27_A_Part_5     98.252031
Q27_A_Part_6     98.883456
Q27_A_Part_7     98.941208
Q27_A_Part_8     98.498441
Q27_A_Part_9     99.002811
Q27_A_Part_10    99.337774
Q27_A_Part_11    89.646941
Q27_A_OTHER      98.702499
dtype: float64

In [ ]:
df['Q28'].value_counts()/df.shape[0]*100

 Amazon Web Services (AWS)                                 3.218727
 Google Cloud Platform (GCP)                               2.841412
They all had a similarly enjoyable developer experience    2.718207
 Microsoft Azure                                           1.747969
None were satisfactory                                     0.365764
 IBM Cloud / Red Hat                                       0.319563
 Oracle Cloud                                              0.231009
 Alibaba Cloud                                             0.130905
Other                                                      0.119355
 VMware Cloud                                              0.115505
 SAP Cloud                                                 0.088553
 Salesforce Cloud                                          0.088553
 Tencent Cloud                                             0.053902
Name: Q28, dtype: float64

In [ ]:
df['IDE'].value_counts() #Sacamos los valores únicos de la columna IDE

Jupyter Notebook                                                                                                                   2341
Visual Studio Code (VSCode) ,Jupyter Notebook                                                                                      1412
Visual Studio Code (VSCode)                                                                                                         939
PyCharm ,Jupyter Notebook                                                                                                           820
Visual Studio Code (VSCode) ,PyCharm ,Jupyter Notebook                                                                              554
                                                                                                                                   ... 
RStudio ,Visual Studio , Spyder, Vim / Emacs,Jupyter Notebook                                                                         1
Jupyter (JupyterLab, Jupyter Notebooks, etc) ,Vi

In [38]:
df_lenguaje = df['lenguaje_utilizado'].str.split(',',expand=True) #Creamos el df_lenguaje con los valores de la columna lenguaje_utilizado separado por columnas
df_lenguaje

0     1            2      3     4     5     6     7     8     9   \
index                                                                         
0      Python     R         None   None  None  None  None  None  None  None   
1         SQL     C          C++   Java  None  None  None  None  None  None   
2      Python   C++         Java   None  None  None  None  None  None  None   
3      Python  None         None   None  None  None  None  None  None  None   
4      Python     C       MATLAB   None  None  None  None  None  None  None   
...       ...   ...          ...    ...   ...   ...   ...   ...   ...   ...   
25968  Python   SQL   Javascript   Bash  None  None  None  None  None  None   
25969  Python  None         None   None  None  None  None  None  None  None   
25970     NaN   NaN          NaN    NaN   NaN   NaN   NaN   NaN   NaN   NaN   
25971  Python   SQL         None   None  None  None  None  None  None  None   
25972     NaN   NaN          NaN    NaN   NaN   NaN   NaN   NaN   NaN   NaN   

         10    11  
index              
0      None  None  
1      None  None  
2      None  None  
3      None  None  
4      None  None  
...     ...   ...  
25968  None  None  
25969  None  None  
25970   NaN   NaN  
25971  None  None  
25972   NaN   NaN  

[25973 rows x 12 columns]

In [39]:
#Sacamos los valores únicos de df_lenguaje
df_lenguaje.stack().value_counts()

Python         21860
 SQL           10756
 C++            5535
 R              5334
 Java           4769
 C              4709
 Javascript     4332
 MATLAB         2935
 Other          2575
 Bash           2216
 None            319
 Julia           305
 Swift           242
dtype: int64

In [36]:
df_IDE = df['IDE'].str.split(',',expand=True) #Creamos el df_lenguaje con los valores de la columna lenguaje_utilizado separado por columnas
df_IDE.head()

0                 1                 2     3     4     5   \
index                                                                       
0       Vim / Emacs              None              None  None  None  None   
1         Notepad++  Jupyter Notebook              None  None  None  None   
2          PyCharm   Jupyter Notebook             Other  None  None  None   
3            Spyder  Jupyter Notebook              None  None  None  None   
4            Spyder           MATLAB   Jupyter Notebook  None  None  None   

         6     7     8     9     10    11    12    13  
index                                                  
0      None  None  None  None  None  None  None  None  
1      None  None  None  None  None  None  None  None  
2      None  None  None  None  None  None  None  None  
3      None  None  None  None  None  None  None  None  
4      None  None  None  None  None  None  None  None

In [37]:
# Contamos los valores únicos del df_IDE -- pendiente eliminar Jupyter repetido
df_IDE.stack().value_counts()

Jupyter Notebook                16233
Visual Studio Code (VSCode)     10040
PyCharm                          7468
Jupyter (JupyterLab              5488
 Jupyter Notebooks               5488
 etc)                            5488
RStudio                          4771
Visual Studio                    4110
 Notepad++                       3937
 Spyder                          3794
 Sublime Text                    2839
MATLAB                           2203
 Vim / Emacs                     1646
 Other                           1491
 None                             526
dtype: int64